app/web/src/workers/webworker.ts

```diff
@@ -401,65 +418,67 @@ const workspaceAtomExistsOnIndexes = (
  * @param meta the new and previous indexes for the changeset.
  * @param fromIndexChecksum the checksum the changeset currently has in the frontend
  */
-const newChangesetIndex = (
+const newChangesetIndex = async (
   db: Database,
   meta: WorkspaceAtomMeta,
   fromIndexChecksum: string | undefined,
 ) => {
   //
   // Create a new empty index
   //
```

Comment:
NOTE: I put a single writeLock around three SQL statements:
- INSERT
- SELECT
- INSERT

Because I didn't want anything hinky to happen switching between locks, and having the first INSERT completed without the second doesn't add any value to the application, so we just block for it all.